utils/puppeteerScrapper.js

```diff
@@ -1,151 +1,90 @@
-      return elements.map(elem => {
-        return elem.getAttribute('src');
-      })
-    });
-
-    waifu.name = name;
-    waifu.bioUrl = `https://mywaifulist.moe${href}`;
-    waifu.imageUrl = image;
-    await Promise.all(promises);
-  }
-  {
-    const targetPage = page;
-    console.info("Waiting the image to be ready");
-    const element = await waitForSelectors([[imageSelector]], targetPage, { timeout, visible: true });
-    console.info("Clicking the image");
-    await element.click({
-      offset: {
-        x: 200,
-        y: 209,
-      },
-    });
-
-    console.info("Waiting the Description to be ready");
-    await waitForSelectors([[descriptionSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting Description of the waifu");
-    const [extract] = await targetPage.$$eval(descriptionSelector, elements => {
-      return elements.map(elem => {
-        return elem.textContent;
-      })
-    });
-    waifu.extract = extract;
-
-    console.info("Waiting the anime where the waifu appears to be ready");
-    await waitForSelectors([[appearsSelector]], targetPage, { timeout, visible: true });
-    console.info("Getting the anime where the waifu appears");
-    const [appearsIn] = await targetPage.$$eval(appearsSelector, elements => {
-      return elements.map(elem => {
-        return {
-          url: `https://mywaifulist.moe${elem.getAttribute('href')}`,
-          text: elem.textContent
-        };
-      })
-    });
-
-    waifu.appearsIn = appearsIn;
+  const nameElement = await page.$(nameSelector)
+  const name = await page.evaluate(el => el.textContent, nameElement)
+  console.info("Name: OK...")
 
-  }
+  const imgElement = await page.$(`[alt="${name.trim()}"]`)
+  const imageUrl = await page.evaluate(el => el.src, imgElement)
+  console.info("Image: OK...")
 
-  console.info("Closing the browser");
-  await browser.close();
+  const whereElement = await page.$(appearsSelector)
+  const source = await page.evaluate(el => el.textContent, whereElement)
+  console.info("Source: OK...")
 
-  return waifu;
+  const popElement = await page.$(popularitySelector)
+  const popularity = await page.evaluate(el => el.textContent.split('#')[1], popElement)
+  console.info("Popularity: OK...")
 
+  const ageElement = await page.$(ageSelector)
+  const age = await page.evaluate(el => el.textContent, ageElement)
+  console.info("Age: OK...")
+
+  console.info("Scrapping complete, closing the browser...");
+  await browser.close();
+
+  return {
+    name,
+    imageUrl,
+    bioUrl,
+    source,
+    popularity,
+    age
+  }
 };
 
+const intializeScrapper = async () => {
+  console.info("Intializing Scrapper...")
+  const browser = await puppeteer.launch();
+
+  const page = await browser.newPage();
+  page.setUserAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36");
+  const timeout = 5000;
```

Comment:
There's nothing that will change this value, so there's no need to create it every time this function is being called. Move it outside the function and on the top of this file with a more explicit name, ie: `PUPETEER_TIMEOUT`